CmrUI/Screens/News Feed tab/NewsFeedVC.swift

```diff
@@ -0,0 +1,333 @@
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if type(of: segue.destination) == NewsFeedDetailedVC.self {
+            let feedDetailedVC = (segue.destination as! NewsFeedDetailedVC)
+            let item  = dataSource[(transitionIndexPath?.row)!]
+            feedDetailedVC.feedItem = item
+            //feedDetailedVC.transitioningDelegate = self
+            feedDetailedVC.modalPresentationStyle = UIModalPresentationStyle.custom
+            feedDetailedVC.navigationItem.hidesBackButton = true
+        }
+    }
+
+    // MARK: UICollectionViewDataSource
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 3
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return dataSource.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if dataSource[indexPath.item].feedType == FeedType.feedImage {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants().kReuseCellID, for: indexPath) as! FeedNewsCell
+            
+            cell.backgroundImageView.image = dataSource[indexPath.item].image
+            cell.titleTextView.text = dataSource[indexPath.item].title
+            cell.subtitleLabel.text = dataSource[indexPath.item].subtitle
+            cell.backgroundColor = UIColor.red
+            cell.isUserInteractionEnabled = true
+            cell.titleTextView.isUserInteractionEnabled = false
+            return cell
+        }
+        else {
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants().kReuseCellTextID, for: indexPath) as! FeedNewsTextCell
+            cell.titleTextView.text = dataSource[indexPath.item].title
+            cell.subtitleLabel.text = dataSource[indexPath.item].subtitle
+            cell.isUserInteractionEnabled = true
+            cell.titleTextView.isUserInteractionEnabled = false
+            return cell
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, willDisplaySupplementaryView view: UICollectionReusableView, forElementKind elementKind: String, at indexPath: IndexPath) {
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        if let supplementaryView = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionElementKindSectionHeader, withReuseIdentifier: Constants().kReuseHeader, for: indexPath) as? FeedNewsSectionHeader {
+            // Configure Supplementary View
+            supplementaryView.backgroundColor = .random()
+            supplementaryView.titleHeader.text = "WEDNESDAY \(indexPath.section + 1) JUNE "
```

Comment:
@vmsavka 1. To see what it looks like, is it possible to see several indexPaths? For example, have several items be in a different day? 
2. The date must be taken from the dataSource. It is an attribute of the FeedItem object.

Reply:
we can add arbitrary count of supplementaryViews
but I decided to replace it by UITableViewController with the same cells. It has build-in vertical scrolling, supports template pattern to make cells reusable, has header and footer views and it is more convenient add an remove any cell

Reply:
This is not what I meant. 
The header text needs to be a grouping of all the feed items, which is done by using their FeedItem.time.